crowd_anki/config/config_settings.py

```diff
@@ -0,0 +1,62 @@
+from collections import namedtuple
+from enum import Enum
+
+from aqt import mw
+from ..utils.constants import USER_FILES_PATH
+
+ConfigEntry = namedtuple("ConfigEntry", ["config_name", "default_value"])
+
+
+class NoteSortingMethods(Enum):
+    FIELD1 = "field1"
+    FIELD2 = "field2"
+    NO_SORTING = "none"
+    GUID = "guid"
+    FLAG = "flag"
+    TAG = "tag"
+    NOTE_MODEL = "notemodel"
+    NOTE_MODEL_ID = "notemodelid"
+
+
+class ConfigSettings:
+    class Properties(Enum):
+        SNAPSHOT_PATH = ConfigEntry("snapshot_path", str(USER_FILES_PATH.resolve()))
+        AUTOMATED_SNAPSHOT = ConfigEntry("automated_snapshot", False)
+        SNAPSHOT_ROOT_DECKS = ConfigEntry("snapshot_root_decks", [])
+        EXPORT_NOTE_SORT_METHODS = ConfigEntry("export_note_sort_methods", [NoteSortingMethods.NO_SORTING.value])
+        EXPORT_NOTES_REVERSE_ORDER = ConfigEntry("export_notes_reverse_order", False)
+
+    def __init__(self):
+        self._config = mw.addonManager.getConfig(__name__)
+        self.load_values()
+
+    def _get(self, prop: Properties):
+        return self._config.get(prop.value.config_name, prop.value.default_value)
+
+    def load_values(self):
+        for prop in self.Properties:
+            setattr(self, prop.value.config_name, self._get(prop))
+
+    def save_values_to_anki(self):
+        for prop in self.Properties:
+            self._config.update({prop.value.config_name: getattr(self, prop.value.config_name)})
```

Comment:
why not just use the dict set value syntax, like:

`self._config.update[prop.value.config_name] = getattr(self, prop.value.config_name)`